### PR TITLE
[FEATURE] Implemented locale manager

### PR DESCRIPTION
**Pull request by trickreich**

### CODE REVIEW

File: src/Sulu/Bundle/Sales/CoreBundle/Manager/LocaleManager.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\Sales\CoreBundle\Manager;
+
+use Sulu\Component\Security\Authentication\UserInterface;
+
+class LocaleManager
```
Comment (turbo-ele) [inline src/Sulu/Bundle/Sales/CoreBundle/Manager/LocaleManager.php:18]: class description

File: src/Sulu/Bundle/Sales/CoreBundle/Manager/LocaleManager.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+     * Function returns the locale that should be used by default.
+     * If request-locale is set, then use this one.
+     * Else return the locale of the user.
+     *
+     * @param UserInterface $user
+     * @param null|string $requestLocale
+     *
+     * @return string
+     */
+    public function retrieveLocale(UserInterface $user, $requestLocale = null)
```
Comment (turbo-ele) [inline src/Sulu/Bundle/Sales/CoreBundle/Manager/LocaleManager.php]: what if user is not logged in?

File: src/Sulu/Bundle/Sales/OrderBundle/Controller/OrderController.php [hunk 3]
```diff
@@ -14,6 +14,7 @@
 use FOS\RestBundle\Controller\Annotations\Post;
 use Hateoas\Representation\CollectionRepresentation;
 use JMS\Serializer\SerializationContext;
+use Sulu\Bundle\Sales\CoreBundle\Manager\LocaleManager;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
 use Sulu\Component\Rest\Exception\EntityNotFoundException;
```
Comment (turbo-ele) [inline src/Sulu/Bundle/Sales/OrderBundle/Controller/OrderController.php:27]: order use statements

File: src/Sulu/Bundle/Sales/ShippingBundle/Controller/ShippingController.php [hunk 4]
```diff
@@ -56,14 +57,22 @@ private function getManager()
     }
 
     /**
+     * @return LocaleManager
+     */
+    private function getLocaleManager()
```
Comment (turbo-ele) [inline src/Sulu/Bundle/Sales/ShippingBundle/Controller/ShippingController.php]: move to bottom (public before private)

File: src/Sulu/Bundle/Sales/OrderBundle/Controller/OrderController.php [hunk 5]
```diff
@@ -60,6 +61,14 @@ private function getDependencyManager()
     }
 
     /**
+     * @return LocaleManager
+     */
+    private function getLocaleManager()
```
Comment (turbo-ele) [inline src/Sulu/Bundle/Sales/OrderBundle/Controller/OrderController.php]: move to bottom (public before private)

**Review by turbo-ele — COMMENTED** [review]
Add build files as well.
Else ack

File: src/Sulu/Bundle/Sales/CoreBundle/Manager/LocaleManager.php (hunk 2 above)
Comment (trickreich) [reply src/Sulu/Bundle/Sales/CoreBundle/Manager/LocaleManager.php]: hmm.. i think this isn't possible?!

File: composer.json [hunk 6]
```diff
@@ -21,8 +21,8 @@
         "oro/doctrine-extensions": "1.0.*",
         "symfony-cmf/routing-bundle": "1.2.*",
 
-        "sulu/sulu": "~1.2",
-        "sulu/product-bundle": "~0.12",
+        "sulu/sulu": "1.3.*",
```
Comment (turbo-ele) [inline composer.json]: what about sulu 1.2.* ?

**Pull request merged**